packages/data-stores/src/site/actions.ts

```diff
@@ -285,64 +285,78 @@ export function createActions( clientCreds: WpcomClientCredentials ) {
-		const { theme, recipe } = selectedDesign;
-
+	function* setThemeOnSite(
+		siteSlug: string,
+		theme: string,
+		styleVariationSlug?: string,
+		keepHomepage = true
+	) {
 		yield wpcomRequest( {
 			path: `/sites/${ siteSlug }/themes/mine`,
 			apiVersion: '1.1',
 			body: {
-				theme: recipe?.stylesheet?.split( '/' )[ 1 ] || theme,
-				style_variation_slug: options?.styleVariation?.slug,
-				dont_change_homepage: true,
+				theme: theme,
+				style_variation_slug: styleVariationSlug,
+				dont_change_homepage: keepHomepage,
 			},
 			method: 'POST',
 		} );
+	}
+
+	function* runThemeSetupOnSite(
+		siteSlug: string,
+		selectedDesign: Design,
+		options?: DesignOptions
+	) {
+		const { recipe, verticalizable } = selectedDesign;
 
 		/*
 		 * Anchor themes are set up directly via Headstart on the server side
 		 * so exclude them from theme setup.
 		 */
 		const anchorDesigns = [ 'hannah', 'gilbert', 'riley' ];
-		if ( anchorDesigns.indexOf( selectedDesign.template ) < 0 ) {
-			const themeSetupOptions: ThemeSetupOptions = {
-				trim_content: true,
-			};
-
-			if ( selectedDesign.verticalizable ) {
-				themeSetupOptions.vertical_id = options?.verticalId;
-			}
+		if ( anchorDesigns.indexOf( selectedDesign.template ) >= 0 ) {
+			return;
+		}
 
-			if ( recipe?.pattern_ids ) {
-				themeSetupOptions.pattern_ids = recipe?.pattern_ids;
-			}
+		const themeSetupOptions: ThemeSetupOptions = {
+			trim_content: true,
+		};
 
-			if ( recipe?.header_pattern_ids ) {
-				themeSetupOptions.header_pattern_ids = recipe?.header_pattern_ids;
-			}
+		if ( verticalizable ) {
+			themeSetupOptions.vertical_id = options?.verticalId;
+		}
 
-			if ( recipe?.footer_pattern_ids ) {
-				themeSetupOptions.footer_pattern_ids = recipe?.footer_pattern_ids;
-			}
+		if ( recipe?.pattern_ids ) {
+			themeSetupOptions.pattern_ids = recipe?.pattern_ids;
+		}
 
-			const response: { blog: string } = yield wpcomRequest( {
-				path: `/sites/${ encodeURIComponent( siteSlug ) }/theme-setup`,
-				apiNamespace: 'wpcom/v2',
-				body: themeSetupOptions,
-				method: 'POST',
-			} );
+		if ( recipe?.header_pattern_ids ) {
+			themeSetupOptions.header_pattern_ids = recipe?.header_pattern_ids;
+		}
 
-			return response;
+		if ( recipe?.footer_pattern_ids ) {
+			themeSetupOptions.footer_pattern_ids = recipe?.footer_pattern_ids;
 		}
+
+		const response: { blog: string } = yield wpcomRequest( {
+			path: `/sites/${ encodeURIComponent( siteSlug ) }/theme-setup`,
+			apiNamespace: 'wpcom/v2',
+			body: themeSetupOptions,
+			method: 'POST',
+		} );
+
+		return response;
+	}
+
+	function* setDesignOnSite( siteSlug: string, selectedDesign: Design, options?: DesignOptions ) {
+		yield* setThemeOnSite(
+			siteSlug,
+			selectedDesign.recipe?.stylesheet?.split( '/' )[ 1 ] || selectedDesign.theme,
+			options?.styleVariation?.slug
+		);
+
+		return yield* runThemeSetupOnSite( siteSlug, selectedDesign, options );
```

Comment:
[nit] I believe `return` here is redundant as the generated value is `yield`-ed already

Reply:
No. The `return` syntax is required. Otherwise, we cannot get the response